dwarf.c

```diff
@@ -1627,91 +1626,100 @@ resolve_unit_addrs_overlap_walk (struct backtrace_state *state,
   struct unit_addrs *new_addrs;
   size_t from;
   size_t to;
+  size_t enclosing_count;
 
   old_addrs = (struct unit_addrs *) old_vec->vec.base;
   old_count = old_vec->count;
   new_addrs = (struct unit_addrs *) new_vec->vec.base;
 
-  for (from = *pfrom, to = *pto; from < old_count; from++, to++)
+  enclosing_count = 0;
+
+  to = 0;
+  for (from = 0; from < old_count; from++)
     {
-      /* If we are in the scope of a larger range that can no longer
-	 cover any further ranges, return back to the caller.  */
+      struct unit_addrs *current_enclosing;
+      new_addrs[to] = old_addrs[from];
+      to++;
+
+      /* While we are in the scope of a larger range that can no longer
+      	 cover any further ranges, pop it from the enclosing stack.  */
+      while (enclosing_count > 0
+        && ((struct unit_addrs**)enclosing->base)[enclosing_count-1]->high <= old_addrs[from].low)
+        {
+          enclosing_count--;
+          enclosing->alc += sizeof (struct unit_addrs*);
```

Comment:
What I want to do here is `enclosing.pop_back()`. I have found `backtrace_vector_grow`, but didn't find reverse function.